multigeiger/multigeiger.ino

```diff
@@ -463,28 +473,42 @@ void loop()
     last_GMZ_counts = GMZ_counts;                              // notice old value
   }
 
+  // Pulse the high voltage at minimum every second
+  #define HVPULSRATE 1000
+  if((millis() - time2hvpuls) >= HVPULSRATE ) {
+    HV_pulse_count = jb_HV_gen_charge__chargepules();       // Charge HV Capacitor - restarts time2hvpusl !
+    hvpulscnt2send += HV_pulse_count;                       // count for sending
+  }
 
-
+  #define DISPLAYREFRESH 10000
+  #define MAXCOUNTS 100
   // Check if there are enough pules detected or if enough time has elapsed. If yes than its
   // time to charge the HV-capacitor and calculate the pulse rate
-  if ((GMZ_counts>=100) || ((count_timestamp - last_count_timestamp)>=10000)) {
+  if ((GMZ_counts >= MAXCOUNTS) || ((millis() - time2display) >= DISPLAYREFRESH )) {
     noInterrupts();
-    isr_GMZ_counts           = 0;
-    interrupts();                                   // initialize ISR values
-    time_difference = count_timestamp - last_count_timestamp;          // Calculate all derived values
-    last_count_timestamp     = count_timestamp;                        // notice the old timestamp
-    HV_pulse_count           = jb_HV_gen_charge__chargepules();        // Charge HV Capacitor
-    accumulated_time         += time_difference;                       // accumulate all the time
-    accumulated_GMZ_counts   += GMZ_counts;                            // accumulate all the pulses
-    lastMinuteLogCounts      += GMZ_counts;
-
-    Count_Rate = (float)GMZ_counts*1000.0/(float)time_difference;      // calculate the current coutrate
-    Dose_Rate = Count_Rate *GMZ_factor_uSvph;                          // ... and dose rate
-
-                                                                       // calculate the radiation over the complete
-                                                                       // time from start
-    accumulated_Count_Rate   = (float)accumulated_GMZ_counts*1000.0/(float)accumulated_time;
-    accumulated_Dose_Rate    = accumulated_Count_Rate *GMZ_factor_uSvph;
+    isr_GMZ_counts = 0;
+    interrupts();                 
+    time2display = millis();
+    time_difference = count_timestamp - last_count_timestamp; // Calculate all derived values
+    last_count_timestamp = count_timestamp;                 // notice the old timestamp
+    HV_pulse_count = jb_HV_gen_charge__chargepules();       // Charge HV Capacitor
+    hvpulscnt2send += HV_pulse_count;                       // count for sending
+    accumulated_time += time_difference;                    // accumulate all the time
+    accumulated_GMZ_counts += GMZ_counts;                   // accumulate all the pulses
+    lastMinuteLogCounts += GMZ_counts;
+
+    Count_Rate = 0;
+    if (time_difference != 0.0) {
+      Count_Rate = (float)GMZ_counts*1000.0/(float)time_difference;  // calculate the current coutrate
+    }
+    Dose_Rate = Count_Rate *GMZ_factor_uSvph;               // ... and dose rate
```

Comment:
and above you added the comment that the conversion factor is for CPM to uSv/h.

maybe we should just get rid of that comment and name the variables like:

`cps_to_uSvph` (or `cpm_...` depending on how the factor is computed).